src/api/use-cart.ts

```diff
@@ -0,0 +1,110 @@
+import type { CartItem } from './types';
+import { useEcomAPI } from './ecom-api-context-provider';
+import { cart as cartType } from '@wix/ecom';
+import { getCartItemSubtotal } from './cart-helpers';
+import {
+    useAddToCart,
+    useCartAndTotals,
+    useRemoveItemFromCart,
+    useUpdateCartItemQuantity,
+} from './api-hooks';
+import { useEffect, useState } from 'react';
+import { media } from '@wix/sdk';
+
+export interface UseCartResult {
+    cartItems: CartItem[];
+    priceSummary?: cartType.PriceSummary;
+
+    isCartMutating: boolean;
+    isAddingToCart: boolean;
+
+    triggerCartUpdate: () => void;
+    triggerAddToCart: (args1: any, args2: any) => void; //  any temporarily
+    handleCheckout: () => void;
+}
+
+export const useCart = (): UseCartResult => {
+    const ecomAPI = useEcomAPI();
+    const {
+        data: cartData,
+        isMutating: isCartMutating,
+        trigger: triggerCartUpdate,
+    } = useCartAndTotals();
+    const cart = cartData?.cart;
+    const cartTotals = cartData?.totals;
+
+    const [cartItems, setCartItems] = useState<CartItem[]>([]);
+
+    const {
+        trigger: triggerAddToCart,
+        data: cartAfterItemAdded,
+        isMutating: isAddingToCart,
+    } = useAddToCart();
+
+    const {
+        trigger: removeItem,
+        data: cartAfterItemRemoved,
+        isMutating: isRemovingItem,
+    } = useRemoveItemFromCart();
+
+    const {
+        trigger: updateItemQuantity,
+        data: cartAfterQuantityUpdated,
+        isMutating: isUpdatingItemQuantity,
+    } = useUpdateCartItemQuantity();
+
+    useEffect(() => {
+        if (cartAfterItemRemoved || cartAfterQuantityUpdated || cartAfterItemAdded) {
+            triggerCartUpdate();
```

Comment:
I see that we get updated cart after adding, removing, updating cart item. So we don't need to fetch it. We only need to make request for cartTotal. 
Will fix tomorrow.